### PR TITLE
fixes #3113: bring in map view in absence of -ObjC linker flag for IB

### LINKED ISSUE #3113: Interface Builder doesn't recognize classes without -ObjC linker flag
Discovered by @friedbunny and applies to any manual binary install method. You can't use `MGLMapView` from IB without referencing it elsewhere in code. 

Should be able to fix this with a combo of `+load` and calling `MGLMapView` from our category loader from #2967. 

/cc @friedbunny @1ec5

### PR DESCRIPTION
**Pull request by incanus**
@friedbunny @1ec5

### CODE REVIEW

File: platform/ios/MGLCategoryLoader.m [hunk 1]
```diff
@@ -4,7 +4,16 @@
...
 #import "NSString+MGLAdditions.h"
 
+#import "MGLMapView.h"
+
 // https://github.com/mapbox/mapbox-gl-native/issues/2966
+// https://github.com/mapbox/mapbox-gl-native/issues/3113
+
+@interface MGLMapView (MGLCategoryLoader)
+
++ (void)mgl_linkMapView;
```
Comment (1ec5) [inline platform/ios/MGLCategoryLoader.m]: It should be possible to just call an NSObject class method. For example, `[MGLMapView description];`.

Comment (incanus) [inline platform/ios/MGLCategoryLoader.m]: Oh, good call. I was wondering about optimization there, and originally went with `+[UIView requiresConstraintBasedLayout]`, but something from `NSObject` would be safer.

Comment (1ec5) [inline platform/ios/MGLCategoryLoader.m]: Come to think of it, ObjC dynamic message dispatch would probably prevent any optimization around whatever class method you’d call here, but I feel better about a method that has to return a string than one that really is a no-op.

**Comment by friedbunny** [comment]
Still seeing `Unknown class MGLMapView in Interface Builder file.` when trying to use this framework with an IB-only project.

**Comment by incanus** [comment]
Hmm, can you set a symbolic breakpoint on `+load` or maybe `+[MGLAccountManager load]`? Since you rolled the binary locally, you should still get symbols.

**Comment by friedbunny** [comment]
Symbolic breakpoint for `+[MGLAccountManager load]` never trips.

**Comment by incanus** [comment]
Hmm, and it does with a code-based path? I'm not as familiar with `+load` as @1ec5, but maybe `awakeFromNib` is more appropriate here?

**Comment by incanus** [comment]
Eh, `awakeFromNib` would be a :chicken: & :white_circle: problem here, since nothing would be in the NIB/XIB to awake. Hmm.

**Comment by incanus** [comment]
We don't yet have a fix here.

**Comment by incanus** [comment]
While this doesn't solve the IB problem, it's still better code & docs and should go into 3.0.1. We're bringing back the `-ObjC` flag, anyway.

**Pull request merged**